src/vm/memory/segments.zig

```diff
@@ -142,6 +144,22 @@ pub const MemorySegmentManager = struct {
         return self.segment_used_sizes.get(index) orelse null;
     }
 
+    /// Retrieves the size of a memory segment by its index if available, else computes it.
+    ///
+    /// This function attempts to retrieve the size of a memory segment by its index. If the size
+    /// is not available in the segment sizes map, it calculates the effective size and returns it.
+    ///
+    /// # Parameters
+    ///
+    /// - `index` (u32): The index of the memory segment.
+    ///
+    /// # Returns
+    ///
+    /// A `u32` representing the size of the segment or a computed effective size if not available.
+    pub fn getSegmentSize(self: *Self, index: u32) ?u32 {
```

Comment:
Is there any particular reason to move this one?

Reply:
I just thought the order was better